home-exercises/src/main/java/extrapractice/bta/LetterCounter.java

```diff
@@ -0,0 +1,46 @@
+package extrapractice.bta;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.stream.Collectors;
+
+public class LetterCounter {
+    static File f = new File("resources/text_file_1000_chars.txt");
+
+    public static void main(String[] args) {
+        HashMap<String, Integer> letterMap = readChars(f);
+        System.out.println(mapToString(letterMap));
+    }
+
+
+    public static HashMap<String, Integer> readChars(File f) {
+        HashMap<String, Integer> letterMap = new HashMap<>();
+        try {
+            String line;
+            BufferedReader reader = new BufferedReader(new FileReader(f));
+            while ((line = reader.readLine()) != null) {
+                String[] split = line.replaceAll("[^a-zA-Z]", "").toUpperCase().split("");
+                if (line.equals("")) {
+                    continue;
+                }
+                for (String s : split) {
+                    if (!letterMap.containsKey(s)) {
```

Comment:
Lai aizstātu šo IF/ELSE ieteiktu uzmest aci map iebūvētajai metodei .merge :
https://nurkiewicz.com/2019/03/mapmerge-one-method-to-rule-them-all.html
https://www.programiz.com/java-programming/library/hashmap/merge
(vari pameklēt arī citus piemērus)